oeps/oep-0047-bp-semantic-versioning.rst

```diff
@@ -0,0 +1,90 @@
+###########################
+OEP-47: Semantic Versioning
+###########################
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0047 </oeps/oep-0047>`
+   * - Title
+     - Semantic Versioning
+   * - Last Modified
+     - 2020-05-08
+   * - Authors
+     - Robert Raposa <rraposa@edx.org>
+   * - Arbiter
+     - Régis Behmo <regis@behmo.com>
+   * - Status
+     - Under Review
+   * - Type
+     - Best Practice
+   * - Created
+     - 2020-03-19
+   * - Review Period
+     - 2020-05-11 - 2020-05-25
+
+.. contents::
+   :local:
+   :depth: 3
+
+Summary
+=======
+
+For any Open edX repository that publishes versioned packages, this OEP recommends following the `Semantic Versioning v2`_.
+
+The most relevant part from the summary of the `Semantic Versioning v2`_ specification:
+
+    Given a version number MAJOR.MINOR.PATCH, increment the:
+
+    * *MAJOR* version when you make *incompatible API changes*,
+    * *MINOR* version when you *add functionality in a backwards compatible manner*, and
+    * *PATCH* version when you make *backwards compatible bug fixes*.
+
+Some additional tips are called out in `Backward Incompatible Changes`_ and `Tooling and Automation`_.
+
+.. note::
+
+    This OEP does not apply to repositories without versions. For example, services like edx-platform, ecommerce, etc. do not have versions.
+
+Context
+=======
+
+Many or all of the Open edX installable libraries already use versions similar to the `Semantic Versioning v2`_ specification. This OEP simply codifies this best practice, and provides additional guidance.
```

Comment:
Is https://github.com/edx/edx-enterprise/releases or https://github.com/edx/edx-ora2/releases a good example of this and are there some libraries that partially follow this OEP but would need to change their release workflow to comply with it?

Reply:
This isn't about changing anyone's workflow.  It is mostly about knowing when to increment X, Y or Z in X.Y.X.  Some people use it more like LARGE.MEDIUM.SMALL in terms of size of change, and that isn't following the semantic versioning.